ccip-sdk/src/evm/index.ts

```diff
@@ -1048,27 +1076,21 @@ export class EVMChain extends Chain<typeof ChainFamily.EVM> {
     const approveTxs = txs.transactions.slice(0, txs.transactions.length - 1)
     let sendTx: TransactionRequest = txs.transactions[txs.transactions.length - 1]
 
-    // approve all tokens (including feeToken, if needed) in parallel
-    let nonce = await this.provider.getTransactionCount(sender)
-    const responses = await Promise.all(
-      approveTxs.map(async (tx: TransactionRequest) => {
-        tx.nonce = nonce++
-        tx = await wallet.populateTransaction(tx)
-        tx.from = undefined
-        const signed = await wallet.signTransaction(tx)
-        const response = await this.provider.broadcastTransaction(signed)
-        this.logger.debug('approve =>', response.hash)
-        return response
-      }),
-    )
-    if (responses.length) await responses[responses.length - 1].wait(1, 60_000) // wait last tx nonce to be mined
+    // Submit approvals sequentially (browser wallets manage nonces internally)
+    let lastApproveResponse: TransactionResponse | undefined
+    for (const tx of approveTxs) {
```

Comment:
This behavior is slightly different than previous; the previous one could send and get multiple approve txs included in parallel, and this one does it serially; please, use the previous approach, with just a simple try/catch over wallet.sendTransaction then sign+broadcast; no need for the specific error check either, as it should be fine to just try both